src/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -331,7 +331,7 @@ public Task<GraphNode<RemoteResolveResult>> WalkAsync(LibraryRange library, NuGe
                 // If we found an exact match then use it.
                 // This allows us to shortcircuit slow feeds even if there's an exact match
                 if (!libraryRange.VersionRange.IsFloating &&
-                    match != null &&
+                    match?.Library != null &&
```

Comment:
Ran into this when trying it out locally. We have a match but it doesn't have a library. Is that a bug in the invoking code? cc @davidfowl